### ISSUE TITLE
mac: AXMathRootRadicand should return a list of children for msqrt

### ISSUE THREAD

**Issue by asurkov** [w3c/mathml-aam#4]
AXMathRootRadicand should expose multiple children in case of <msqrt>child1 child2 child3 ... childN</msqrt>.

See [spec](https://mathml-refresh.github.io/mathml-core/#radicals-msqrt-mroot):

"The children of the <msqrt> element are laid out using the algorithm of the <mrow> element to produce a box that is also called the base."

Original @fred-wang report in https://github.com/w3c/mathml-aam/pull/2_:

Returning first child is indeed what WebKit currently does. However, I wonder whether this is actually broken, AccessibilityMathMLElement::mathRadicandObject refers to https://bugs.webkit.org/show_bug.cgi?id=146452

Maybe AXMathRootRadicand should instead return an array of children? Probably worth asking jcraig@apple.com and cfleizach@apple.com to see what is most convenient for VoiceOver.

**Comment by cookiecrook** [comment]
@asurkov I can't edit your post, but could you remove the email addresses?

**Comment by cookiecrook** [comment]
It's not clear to me if this issue is sufficiently addressed by the discussion in the linked WebKit issue, which was closed early last year. I'd suggest closing this one too. 

If this is another issue that is still unresolved, please reopen or rephrase, with another example. Thanks.

**Comment by fred-wang** [comment]
OK, let me rephrase and clarify:

So first for `mroot`, we can write something like

     `<math><mroot><mrow><mn>1</mn><mo>+</mo><mn>2</mn></mrow><mn>3</mn></mroot></math>`

which is the "cubic root of 'one plus two'" where 3 (second child) is the `AXMathRootIndex` and "(1+2)" (first child) is the `AXMathRootRadicand`.

For square root, we don't need an explicit root index and can just write:

     `<math><msqrt><mrow><mn>1</mn><mo>+</mo><mn>2</mn></mrow></msqrt></math>`

which is the "square root of 'one plus two'". But actually, using an explicit `<mrow>` to force child position/semantics is no longer needed, MathML allows to just write:

     `<math><msqrt><mn>1</mn><mo>+</mo><mn>2</mn></msqrt></math>`

With the current mapping described in the mathml-aam spec, AXMathRootRadicand is now just the first child "1". Before https://bugs.webkit.org/show_bug.cgi?id=146452, VoiceOver used to incorrectly read that "square root of one".

It seems that the bug is fixed and VoiceOver now correctly reads it "square root of 'one plus two'" (it's closed source, so no idea how it was actually fixed). But still, it's a bit weird (from a naming/semantic point of view) that AXMathRootRadicand points to the first child only.

So this is issue is: (1) maybe we should instead return the array of children, similar to what is done with AXMathSubscript/AXMathSupscript (2) Or maybe the attribute is not really useful at all, and we can just remove that AXMathRootRadicand attribute for `<msqrt>`?

**Comment by cookiecrook** [comment]
Summary from the discussion with Apple colleagues: "AXMathRootRadicand [should] return an array of elements. This would cover the root of any index and possible variants like the root of a complex number or even future tweaks to the spec." 

Filed https://webkit.org/b/230986 and assigned to Andres at his request.

**Comment by fred-wang** [comment]
@cookiecrook Thanks. Just to be sure, this also holds for `<mroot>` (even if it will always return an array of one element1)?

**Comment by fred-wang** [comment]
> @cookiecrook Thanks. Just to be sure, this also holds for `<mroot>` (even if it will always return an array of one element)?

Andres replied positively on https://bugs.webkit.org/show_bug.cgi?id=230986#c4
